### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by keiranjprice101**
This is the updated proof of concept - turned initial implementation.

Currently it works only on local configuration, but there are issues opened to address this.

### CODE REVIEW

File: src/app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          flexDirection: "column",
+        }}
+      >
+        <h2>Not Found</h2>
+        <p>Could not find requested resource</p>
```
Comment (Pasarus) [inline src/app/not-found.tsx:13]:
```suggestion
        <p>Could not find the requested resource.</p>
```

File: src/app/[...slug]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  // We expect a route of /instrument_name/experiment_number/filename
+  // This will result in a slug list of [instrument_name, experiment_number, filename]
+  const [instrument, experimentNumber, filename] = params.slug;
+  if (params.slug.length != 3) {
+    notFound();
+  }
+  return (
+    <main className="h5-container">
+      <NexusViewer
+        filepath={`${instrument.toUpperCase()}/RBNumber/RB${experimentNumber}/autoreduced/${filename}`}
```
Comment (Pasarus) [inline src/app/[...slug]/page.tsx:15]: When does the hopefully in future configurable "/ceph" get added? is it the backend?

**Review by Pasarus — CHANGES_REQUESTED** [review]
Quick question and slight wording adjustment

File: src/app/[...slug]/page.tsx (hunk 2 above)
Comment (keiranjprice101) [reply src/app/[...slug]/page.tsx:15]: So the `/ceph` is the base directory where this structure is expected to be and is configured in the backend.

For example, if I had ~/mari/RBNumber/RB123/autoreduced/mynexus.nxspe. The dataviewer requests the filepath `mari/RBNumber/RB123/autoreduced/mynexus.nxspe`.

If my backend is configured with the env var `CEPH_DIR = /Users/keiran` Then it gets the file from `/Users/keiran/mari/RBNumber/RB123/autoreduced/mynexus.nxspe` 

I hope that is somewhat clear

**Review by Pasarus — APPROVED** [review]

File: src/app/[...slug]/page.tsx (hunk 2 above)
Comment (Pasarus) [reply src/app/[...slug]/page.tsx:15]: Clear, thank you! I thought it was worth documenting here.

**Pull request merged**